api/src/main/scala/hmda/api/http/admin/InstitutionAdminHttpApi.scala

```diff
@@ -0,0 +1,82 @@
+package hmda.api.http.admin
+
+import akka.actor.{ ActorRef, ActorSystem }
+import akka.pattern.ask
+import akka.event.LoggingAdapter
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.server.Directives._
+import hmda.api.http.HmdaCustomDirectives
+import hmda.api.protocol.processing.{ ApiErrorProtocol, InstitutionProtocol }
+import hmda.model.institution.Institution
+import hmda.persistence.HmdaSupervisor.FindActorByName
+import hmda.persistence.institutions.InstitutionPersistence
+import hmda.persistence.institutions.InstitutionPersistence.{ CreateInstitution, GetInstitutionById, ModifyInstitution }
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.util.Timeout
+import hmda.api.protocol.admin.WriteInstitutionProtocol
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+trait InstitutionAdminHttpApi
+    extends WriteInstitutionProtocol
+    with ApiErrorProtocol
+    with HmdaCustomDirectives {
+
+  implicit val system: ActorSystem
+  implicit val materializer: ActorMaterializer
+  implicit val timeout: Timeout
+
+  val log: LoggingAdapter
+
+  val institutionsWritePath =
+    path("institutions") {
+      timedPost { uri =>
+        entity(as[Institution]) { institution =>
+          extractExecutionContext { executor =>
+            implicit val ec: ExecutionContext = executor
+            val supervisor = system.actorSelection("/user/supervisor")
+            val fInstitutionsActor = (supervisor ? FindActorByName(InstitutionPersistence.name)).mapTo[ActorRef]
+            val fCreated = for {
+              a <- fInstitutionsActor
+              i <- (a ? CreateInstitution(institution)).mapTo[Institution]
+            } yield i
+
+            onComplete(fCreated) {
+              case Success(i) =>
+                complete(ToResponseMarshallable(StatusCodes.Created -> i))
```

Comment:
I think that it would be good to return a response that informs that user if the institution already exists, if it does. I think that this should be relatively easy to implement using the same pattern as the put.